supervisor/supervisor.go

```diff
@@ -275,8 +277,12 @@ func (s *Supervisor) Start() error {
 		"cpus":        s.machine.Cpus,
 	}).Debug("containerd: supervisor running")
 	go func() {
-		for i := range s.tasks {
-			s.handleTask(i)
+		for {
+			select {
+			case i := <-s.tasks:
```

Comment:
If the parallelism is removed, the `select` is indeed enough IMHO.

I would remove the for and just select on the first 2 cases though:
```go
select {
  case ctx.Done():
      // cancel send task and return error in evt.ErrChan()
      // no need to add the task anymore as it's being canceled by the caller
      return
  case s.tasks <- evt:
      TaskCounter.Add(1)
      return
}
```